tests/pytests/test_index_oom.py

```diff
@@ -221,7 +249,73 @@ def test_oom_query_error(env):
     env.expect(parsed_query).error().equal(f'{idx_name}: '
     'Index background scan failed due to OOM. Queries cannot be executed on an incomplete index.')
 
+  # Test FT.DEBUG FT.SEARCH/AGGREGATE command
+  for query_type in ['SEARCH', 'AGGREGATE']:
+    parsed_query = f'_FT.DEBUG FT.{query_type} {idx_name} * TIMEOUT_AFTER_N 3 DEBUG_PARAMS_COUNT 2 '
+    env.expect(parsed_query).error().equal(f'{idx_name}: '
+    'Index background scan failed due to OOM. Queries cannot be executed on an incomplete index.')
+
   # Verify ft info possible
   env.expect('FT.INFO', idx_name).noError()
-  # Verify ft drop possible
-  env.expect('FT.DROP', idx_name).ok()
+  # Verify ft dropindex possible
+  env.expect('FT.DROPINDEX', idx_name).ok()
+
+def test_cluster_oom(env):
+  conn = getConnectionByEnv(env)
+  n_docs_per_shard = 1000
+  n_docs = n_docs_per_shard * env.shardsCount
+  for i in range(n_docs):
+    res = conn.execute_command('HSET', f'doc{i}', 't', f'text{i}')
+    env.assertEqual(res, 1)
+
+  # Set pause on OOM for all shards
+  pause_on_oom_cmd = ' '.join([bgScanCommand(),' SET_PAUSE_ON_OOM', 'true'])
+  run_command_on_all_shards(env, pause_on_oom_cmd)
+  # Set pause on half of the docs for all shards
+  pause_on_scanned_docs_cmd = ' '.join([bgScanCommand(),' SET_PAUSE_ON_SCANNED_DOCS', str(n_docs_per_shard//50)])
+  run_command_on_all_shards(env, pause_on_scanned_docs_cmd)
+  # Set pause before scan for all shards
+  pause_before_scan_cmd = ' '.join([bgScanCommand(),' SET_PAUSE_BEFORE_SCAN', 'true'])
+  run_command_on_all_shards(env, pause_before_scan_cmd)
+
+  for oom_on_allShards in [True, False]:
+    # Create an index
+    idx_str = f'idx{1 if oom_on_allShards else 2}'
+    res = conn.execute_command('FT.CREATE', idx_str, 'SCHEMA', 'txt', 'TEXT')
+    env.assertEqual(res, 'OK')
+    # Wait for pause before scan
+    allShards_waitForIndexStatus(env, 'NEW', idx_str)
+    # Resume all shards
+    resume_cmd = ' '.join([bgScanCommand(),' SET_BG_INDEX_RESUME'])
+    run_command_on_all_shards(env, resume_cmd)
+    # Wait for pause on docs scanned
+    allShards_waitForIndexPauseScan(env, idx_str)
+    if oom_on_allShards:
+      # Set tight memory limit for all shards
+      allShards_set_tight_maxmemory_for_oom(env)
+      run_command_on_all_shards(env, resume_cmd)
+      # Wait for OOM on all shards
+      allShards_waitForIndexStatus(env, 'PAUSED_ON_OOM', idx_str)
+      # Resume all shards
+      run_command_on_all_shards(env, resume_cmd)
+    else: # OOM on single shard
+      # Set tight memory limit for one shard
+      shard_id =  env.shardsCount
+      shard_set_tight_maxmemory_for_oom(env, shard_id)
+      run_command_on_all_shards(env, resume_cmd)
+      # Wait for OOM on all shards
+      shard_waitForIndexStatus(env, shard_id, 'PAUSED_ON_OOM', idx_str)
+      # Resume all shards
+      env.getConnection(shard_id).execute_command(bgScanCommand(), 'SET_BG_INDEX_RESUME')
```

Comment:
Consider adding a validation over the info of each individual shard as well with `_ft.info` and confirm your assumption in each scenario.
Also consider splitting this into two tests the diff becomes large...

Reply:
Test were split to OOM on all shard and OOM on a single shard, 
_ft.info cannot be called from a specific shard so used info metric to check for OOM.